lib/orocos/process.rb

```diff
@@ -1171,9 +1199,25 @@ def self.wait_running(process, timeout = nil, name_service = Orocos::CORBA.name_
                     block.call(process)
                 else
                     # Get any task name from that specific deployment, and check we
-                    # can access it. If there is none
+                    # can access it either through the ior pipe or through the CORBA name
+                    # service.
+                    if ior_read?
+                        ior_message += ior_read.read_nonblock 1
+                        task_name, ior, msg_status = parse_ior_message ior_message
+                        case msg_status
+                        when :PARTIAL
+                            return false
+                        when :FULL_IOR
+                            process.register_ior(task_name, ior)
+                            return false
+                        end
+                    end
+
                     all_reachable = process.task_names.all? do |task_name|
-                        if name_service.task_reachable?(task_name)
+                        if process.has_ior_for? task_name
+                            Orocos.debug "#{task_name}'s IOR is registered"
+                            true
+                        elsif name_service.task_reachable?(task_name)
                             Orocos.debug "#{task_name} is reachable"
```

Comment:
could a task, that has an IOR registered, be unreachable?

Reply:
No because that is a characteristic of the name service, which we are avoiding by registering the IORs we get from the pipe.